pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -31,14 +32,27 @@ if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
     podman volume create $worker_volume_name
     podman pod create --name $worker_pod_name
     echo "creating init container"
-    podman create \
-      --pod $worker_pod_name \
-      --init-ctr=always \
-      --rm \
-      -v $worker_volume_name:/tmp \
-      $full_kernel_module_image \
-      /bin/sh -c "mkdir -p /tmp/opt/lib/modules && \
-      cp -R /opt/lib/modules/* /tmp/opt/lib/modules"
+    if [[ -z "$FIRMWARE_FILES_PATH" ]]; then
+      podman create \
+            --pod $worker_pod_name \
+            --init-ctr=always \
+            --rm \
+            -v $worker_volume_name:/tmp \
+            $full_kernel_module_image \
+            /bin/sh -c "mkdir -p /tmp/opt/lib/modules && \
+            cp -R /opt/lib/modules/* /tmp/opt/lib/modules"
+    else
+      podman create \
+            --pod $worker_pod_name \
+            --init-ctr=always \
+            --rm \
+            -v $worker_volume_name:/tmp \
+            $full_kernel_module_image \
+            /bin/sh -c "mkdir -p /tmp/opt/lib/modules && \
+            cp -R /opt/lib/modules/* /tmp/opt/lib/modules && \
+            mkdir -p /tmp$firmware_files_path && \
+            cp -R $firmware_files_path/* /tmp$firmware_files_path"
+    fi
```

Comment:
```suggestion
    copycmd='mkdir -p /tmp/opt/lib/modules && cp -R /opt/lib/modules/* /tmp/opt/lib/modules'
    if [[ -n "$FIRMWARE_FILES_PATH" ]]; then
      copycmd+=' mkdir -p /tmp/${firmware_files_path} && cp -R ${firmware_files_path}/* /tmp/${firmware_files_path}'
    fi
    podman create \
          --pod $worker_pod_name \
          --init-ctr=always \
          --rm \
          -v $worker_volume_name:/tmp \
          $full_kernel_module_image \
          /bin/sh -c "${copycmd}"
    ```

Reply:
WDYT about this suggestion?

The changes are:
1. Only add the firmeware files in the `if` statement instead of duplicating the `podman command`
2. Split the command using `;` instead of `&&` because there is no reason to fail the second command just because the first one failed
3. Fixed the directory from `/tmp$firmware_files_path` to `/tmp/$firmware_files_path` - how did it work before the fix?
4. Enclosed the variables with `${}` just because it is safer - there is no real use in it here :)

Reply:
1. Good idea, it looks much cleaner.
2. Understood, I've added `;` to the end of the copycmd `copycmd="mkdir -p /tmp/opt/lib/modules && cp -R /opt/lib/modules/* /tmp/opt/lib/modules;"`
3. For example lets say the firmware files are in dir `/opt/lib/firmware`, so with
 `/tmp$firmware_files_path` it will be `/tmp/opt/lib/firmware`.
 4. Good idea, also I changed `'` to `"`.

Reply:
> For example lets say the firmware files are in dir /opt/lib/firmware, so with
/tmp$firmware_files_path it will be /tmp/opt/lib/firmware.

This is correct, in KMM's code we use `filepath.Join` to always return a clean path - we need to find a bash solution for it.